File: backend/models/sale.go
package models

// Bangladeshi Payment Methods:
// "cash" | "bkash" | "nagad" | "rocket" | "bank_transfer" | "card" | "dbbl_nexus" | "upay"

type Sale struct {
	ID              string     `json:"id"`
	InvoiceNumber   string     `json:"invoice_number"`
	CustomerName    string     `json:"customer_name"`
	CustomerEmail   string     `json:"customer_email"`
	CustomerPhone   string     `json:"customer_phone"`
	CustomerAddress string     `json:"customer_address"`
	Items           []SaleItem `json:"items"`
	SubTotal        float64    `json:"sub_total"`
	DiscountAmount  float64    `json:"discount_amount"`
	TaxAmount       float64    `json:"tax_amount"`
	TotalAmount     float64    `json:"total_amount"`
	PaidAmount      float64    `json:"paid_amount"`
	PaymentMethod   string     `json:"payment_method"`
	TransactionID   string     `json:"transaction_id"` // bKash/Nagad/Rocket/Bank TrxID
	Status          string     `json:"status"`         // "pending" | "completed" | "cancelled" | "refunded"
	Note            string     `json:"note"`
	CreatedBy       string     `json:"created_by"`
	CreatedAt       string     `json:"created_at"`
	UpdatedAt       string     `json:"updated_at"`
}

type SaleItem struct {
	ProductID   string  `json:"product_id"`
	ProductName string  `json:"product_name"`
	SKU         string  `json:"sku"`
	Quantity    int     `json:"quantity"`
	UnitPrice   float64 `json:"unit_price"`
	CostPrice   float64 `json:"cost_price"` // COGS: product cost at time of sale
	Discount    float64 `json:"discount"`
	Subtotal    float64 `json:"subtotal"`
}

File: admin/src/types/sale.ts
export interface SaleItem {
  product_id: string
  product_name: string
  sku: string
  quantity: number
  unit_price: number
  cost_price: number
  discount: number
  subtotal: number
}

export interface Sale {
  id: string
  invoice_number: string
  customer_name: string
  customer_email: string
  customer_phone: string
  customer_address: string
  items: SaleItem[]
  sub_total: number
  discount_amount: number
  tax_amount: number
  total_amount: number
  paid_amount: number
  payment_method: string
  transaction_id: string
  status: 'pending' | 'completed' | 'cancelled' | 'refunded'
  note: string
  created_by: string
  created_at: string
  updated_at: string
}

export interface CreateSaleInput {
  customer_name: string
  customer_email: string
  customer_phone: string
  customer_address: string
  items: { product_id: string; quantity: number; unit_price: number; discount: number }[]
  discount_amount: number
  paid_amount: number
  payment_method: string
  transaction_id: string
  note: string
}
